src/modules/rc_update/RCUpdateTest.cpp

```diff
@@ -38,15 +38,120 @@
 
 using namespace rc_update;
 
-TEST(RCUpdateTest, ModeSlotUnassigned)
+class TestRCUpdate : public RCUpdate
+{
+public:
+	void UpdateManualSwitches(const hrt_abstime &timestamp_sample) { RCUpdate::UpdateManualSwitches(timestamp_sample); }
+	void updateParams() { RCUpdate::updateParams(); }
+	void setChannel(size_t index, float channel_value) { _rc.channels[index] = channel_value; }
+};
```

Comment:
Is this class really needed? The only function it adds is `setChannel` right?

Reply:
Well in https://github.com/PX4/PX4-Autopilot/commit/4bd2d4cf3576b5f87c761a485420a814f3084cec#diff-5c02ad0b2293fe7ee164aea9fe5387fba87e6305fa543acc9f08a6a5cc09716cL93 I removed the private access specifier of member variables to be able to mutate them from the testing. In this pr I'm putting back the protected access specifier and this class is used as kind of a test "friend" explicitly exposing what's needed for testing. The test and class should be restructured to cover all testing without mutating internals. This helper class is an in-between incremental step showing what's missing and can be removed again when doing these changes.

Reply:
Oh gotcha, didn't notice that those were `protected` functions!